app/src/main/java/org/sagebionetworks/research/mpower/medication/MedicationDayFragment.kt

```diff
@@ -0,0 +1,175 @@
+package org.sagebionetworks.research.mpower.medication
+
+import android.app.Activity
+import android.app.AlertDialog
+import android.content.Context
+import android.app.Dialog
+import android.os.Bundle
+import android.support.v7.app.AppCompatDialogFragment
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.dialog_medication_day.day_selection_back
+import kotlinx.android.synthetic.main.dialog_medication_day.day_selection_save
+import kotlinx.android.synthetic.main.dialog_medication_day.day_selection_title
+import kotlinx.android.synthetic.main.dialog_medication_day.medication_day_recycler
+import kotlinx.android.synthetic.main.list_item_day.view.day_text
+import org.sagebionetworks.research.mpower.R
+import org.sagebionetworks.research.mpower.R.layout
+import org.slf4j.LoggerFactory
+
+class MedicationDayFragment: AppCompatDialogFragment() {
+
+    private val LOGGER = LoggerFactory.getLogger(
+            MedicationDayFragment::class.java)
+
+
+    companion object {
+        val ARG_SCHED_ID = "ARG_SCHED_ID"
+        val ARG_NAME = "ARG_NAME"
+        val ARG_TIME = "ARG_TIME"
+        val ARG_DAYS = "ARG_DAYS"
+
+        fun newInstance(id: String, name: String, time: String, days: String): MedicationDayFragment {
+            val fragment = MedicationDayFragment()
+            val args = Bundle()
+            args.putString(ARG_SCHED_ID, id)
+            args.putString(ARG_NAME, name)
+            args.putString(ARG_TIME, time)
+            args.putString(ARG_DAYS, days)
+            fragment.arguments = args
+            return fragment
+        }
+    }
+
+    lateinit var customView: View
+    lateinit var listener: DaySelectedListener
+    var selectedDays: MutableList<String> = arrayListOf()
+
+    lateinit var schedId: String
+    lateinit var name: String
+    lateinit var time: String
+    lateinit var days: String
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if(getArguments() != null) {
+            schedId = arguments!!.getString(ARG_SCHED_ID)
+            name = arguments!!.getString(ARG_NAME)
+            time = arguments!!.getString(ARG_TIME)
+            days = arguments!!.getString(ARG_DAYS)
+
+            selectedDays = days.split(",").toMutableList()
+            val size = selectedDays.size
+            LOGGER.debug("Selected days on init: $size")
+        } else {
+            LOGGER.debug("No arguments found")
```

Comment:
I think this should be a LOGGER.warn() since we don't want someone creating a Fragment like this without the arguments.